tests/Microsoft.Identity.Test.Integration.netcore/WAM/RuntimeBrokerTests.cs

```diff
@@ -135,5 +136,100 @@ public async Task WamInteractiveAuthNoWindowHandleAsync()
             }
 
         }
+
+        [TestMethod]
+        public async Task WamUsernamePasswordRequestAsync()
+        {
+            var labResponse = await LabUserHelper.GetDefaultUserAsync().ConfigureAwait(false);
+            string[] scopes = { "User.Read" };
+
+            IPublicClientApplication pca = PublicClientApplicationBuilder
+               .Create(labResponse.App.AppId)
+               .WithAuthority(labResponse.Lab.Authority, "organizations")
+               .WithBrokerPreview().Build();
+
+            // Acquire token using username password
+            var result = await pca.AcquireTokenByUsernamePassword(scopes, labResponse.User.Upn, new NetworkCredential("", labResponse.User.GetOrFetchPassword()).SecurePassword).ExecuteAsync().ConfigureAwait(false);
+
+            AssertAuthResult(result, TokenSource.Broker, labResponse.Lab.TenantId);
+
+            // Get Accounts
+            var accounts = await pca.GetAccountsAsync().ConfigureAwait(false);
+            Assert.IsNotNull(accounts);
+
+            var account = accounts.FirstOrDefault();
+            Assert.IsNotNull(account);
+
+            // Acquire token silently
+            result = await pca.AcquireTokenSilent(scopes, account).ExecuteAsync().ConfigureAwait(false);
+
+            AssertAuthResult(result, TokenSource.Cache, labResponse.Lab.TenantId);
+
+            // Remove Account
+            await pca.RemoveAsync(account).ConfigureAwait(false);
+
+            // Assert the account is removed
+            accounts = await pca.GetAccountsAsync().ConfigureAwait(false);
+
+            Assert.IsNotNull(accounts);
+            Assert.AreEqual(0, accounts.Count());
+        }
+
+        [TestMethod]
+        public async Task WamUsernamePasswordRequestMsaPassthroughAsync()
```

Comment:
MSA-PT, as the name suggests, is to allow an MSA users to acquire a token for certain client apps which do not natively support MSA.

However: 

1. your test does fetch an MSA user from the lab, it fetches an AAD user
2. MSA does not suport ROPC. In any shape or form.

Pls delete the test